File: AddressBook/Personal.cpp
// Personal.cpp
#include "Personal.h"

Personal::Personal()
	:name(""), address(""), telephoneNumber(""), emailAddress(""){

}
Personal::Personal(string name, string address,
	string telephoneNumber, string emailAddress) {
	this->name = name;
	this->address = address;
	this->telephoneNumber = telephoneNumber;
	this->emailAddress = emailAddress;
}

Personal::~Personal() {
} // ¼Ò¸êÀÚ
Personal::Personal(const Personal& source) {
	this->name = source.name;
	this->address = source.address;
	this->telephoneNumber = source.telephoneNumber;
	this->emailAddress = source.emailAddress;
}

bool Personal::IsEqual(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) == 0 &&
		this->address.compare(other.address) == 0 &&
		this->telephoneNumber.compare(other.telephoneNumber) == 0 &&
		this->emailAddress.compare(other.emailAddress) == 0) {
		ret = true;
	}
	return ret;
}
bool Personal::IsNotEqual(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) != 0 ||
		this->address.compare(other.address) != 0 ||
		this->telephoneNumber.compare(other.telephoneNumber) != 0 ||
		this->emailAddress.compare(other.emailAddress) != 0) {
		ret = true;
	}
	return ret;
}
bool Personal::operator==(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) == 0 &&
		this->address.compare(other.address) == 0 &&
		this->telephoneNumber.compare(other.telephoneNumber) == 0 &&
		this->emailAddress.compare(other.emailAddress) == 0) {
		ret = true;
	}
	return ret;
}
bool Personal::operator!=(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) != 0 ||
		this->address.compare(other.address) != 0 ||
		this->telephoneNumber.compare(other.telephoneNumber) != 0 ||
		this->emailAddress.compare(other.emailAddress) != 0) {
		ret = true;
	}
	return ret;
}
Personal& Personal::operator=(const Personal& other) {
	this->name = other.name;
	this->address = other.address;
	this->telephoneNumber = other.telephoneNumber;
	this->emailAddress = other.emailAddress;
	
	return *this;
} 

File: AddressBook/Java/Personal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package addressbook;
/**
 *
 * @author Taehoo
 */
public class Personal {
    public String name;
    public String address;
    public String telephoneNumber;
    public String emailAddress;
    
    public Personal(){
        this.name = "";
        this.address= "";
        this.telephoneNumber = "";
        this.emailAddress = "";
    }
    public Personal(String name, String address, String telephoneNumber,
            String emailAddress){
        this.name = name;
        this.address = address;
        this.telephoneNumber = telephoneNumber;
        this.emailAddress = emailAddress;
    }
   
    public boolean IsEqual(Personal other){
        boolean ret = false;
        
        if(this.name.compareTo(other.name) ==0 && this.address.compareTo(other.address) ==0 &&
                this.telephoneNumber.compareTo(other.telephoneNumber) == 0 && this.emailAddress.compareTo(other.emailAddress) == 0){
            ret = true;
        } 
        
        return ret;
    } 
    
    public boolean IsNotEqual(Personal other){
        boolean ret = false;
        
       if(this.name.compareTo(other.name) !=0 || this.address.compareTo(other.address) !=0 ||
                this.telephoneNumber.compareTo(other.telephoneNumber) != 0 || this.emailAddress.compareTo(other.emailAddress) != 0){
            ret = true;
        }
       
       return ret;
    }
    
    public String GetName(){
        return this.name;
    }
    public String GetAddress(){
        return this.address;
    }
    public String GetTelephoneNumber(){
        return this.telephoneNumber;
    }
    public String GetEmailAddress(){
        return this.emailAddress;
    }
    
}
